app/src/main/kotlin/com/wire/android/ui/home/newconversation/groupOptions/GroupOptionsScreen.kt

```diff
@@ -82,147 +88,189 @@ fun GroupOptionScreenContent(
+            ErrorDialog(it, onErrorDismissed)
+        }
         if (showAllowGuestsDialog) {
-            WireDialog(
-                title = stringResource(R.string.disable_guests_dialoug_title),
-                text = stringResource(R.string.disable_guests_dialoug_description),
-                onDismiss = onAllowGuestsDialogDismissed,
-                buttonsHorizontalAlignment = false,
-                optionButton1Properties = WireDialogButtonProperties(
-                    onClick = onNotAllowGuestsClicked,
-                    text = stringResource(id = R.string.disable_guests_dialoug_button),
-                    type = WireDialogButtonType.Primary
-                ), optionButton2Properties = WireDialogButtonProperties(
-                    text = stringResource(R.string.allow_guests),
-                    onClick = onAllowGuestsClicked,
-                    type = WireDialogButtonType.Primary
-                ), dismissButtonProperties = WireDialogButtonProperties(
-                    text = stringResource(R.string.label_cancel),
-                    onClick = onAllowGuestsDialogDismissed
-                )
-            )
+            AllowGuestsDialog(onAllowGuestsDialogDismissed, onNotAllowGuestsClicked, onAllowGuestsClicked)
+        }
+    }
+}
+
+@Composable
+private fun GroupOptionState.GroupOptionsScreenMainContent(
+    internalPadding: PaddingValues,
+    onAllowGuestChanged: (Boolean) -> Unit,
+    onAllowServicesChanged: (Boolean) -> Unit,
+    onReadReceiptChanged: (Boolean) -> Unit,
+    onContinuePressed: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(internalPadding)
+            .background(MaterialTheme.colorScheme.background),
+        verticalArrangement = Arrangement.SpaceBetween
+    ) {
+        Column() {
+            AllowGuestsOptions(onAllowGuestChanged)
+            AllowServicesOptions(onAllowServicesChanged)
+            ReadReceiptsOptions(onReadReceiptChanged)
         }
+        ContinueButton(onContinuePressed)
     }
 }
 
+@Composable
+private fun GroupOptionState.ReadReceiptsOptions(onReadReceiptChanged: (Boolean) -> Unit) {
+    GroupConversationOptionsItem(
+        title = stringResource(R.string.read_receipts),
+        switchState = SwitchState.Enabled(value = isReadReceiptEnabled,
+            isOnOffVisible = false,
+            onCheckedChange = { onReadReceiptChanged.invoke(it) }),
+        arrowType = ArrowType.NONE,
+        clickable = Clickable(enabled = false, onClick = {}, onLongClick = {}),
+        modifier = Modifier
+            .fillMaxWidth()
+            .background(MaterialTheme.colorScheme.surface)
+    )
+
+    Text(
+        text = stringResource(R.string.read_receipts_description),
+        fontWeight = FontWeight.Normal,
+        color = MaterialTheme.wireColorScheme.secondaryText,
+        modifier = Modifier.padding(MaterialTheme.wireDimensions.spacing16x),
+        textAlign = TextAlign.Left,
```

Comment:
for texts we use pre defined styles so we don't have to use fontsize , font style and weight, something like 
style = MaterialTheme.wireTypography.body01

Reply:
Giving that:
- This PR is blocking people
- It was already like this, I just moved it around for better readability
- Changing it would trigger another rebuild which takes time

I think we could tackle this in the future.